test/server/controllers/UserController.spec.js

```diff
@@ -154,25 +154,25 @@ describe('UserController', function() {
 
         });
 
-        xit('should use request.params.id and ignore user.id', async function() {
-            const request = {
-                body: user,
+        it('should use request.params.id and ignore user.id', async function() {
+           connection.query.mockReturnValue({rowCount:1, rows: [database[0]]})
+
+           const request = {
+                body: submittedUsers[0],
```

Comment:
You'll want to use `{..submittedUsers[0]}` here,, to clone `submittedUsers[0]` rather than referencing it directly.  Javascript is pass by reference and the `putUser` method overwrites the `password` field on the user submitted with the hash of the password before sending it on to the database.  This doesn't matter in a normal request context, where it's rare for requests to get chained.  But in this context, we want to reuse `submittedUsers` in the other tests, so we don't want `submittedUsers[0].password` to get overwritten.